src/PlaywrightSharp.Firefox/FirefoxPage.cs

```diff
@@ -61,6 +80,198 @@ public Task SetViewportAsync(Viewport viewport)
+                    break;
+                case RuntimeConsoleFirefoxEvent runtimeConsole:
+                    break;
+                case PageDialogOpenedFirefoxEvent pageDialogOpened:
+                    break;
+                case PageBindingCalledFirefoxEvent pageBindingCalled:
+                    break;
+                case PageFileChooserOpenedFirefoxEvent pageFileChooserOpened:
+                    break;
+                case PageWorkerCreatedFirefoxEvent pageWorkerCreated:
+                    OnWorkerCreated(pageWorkerCreated);
+                    break;
+                case PageWorkerDestroyedFirefoxEvent pageWorkerDestroyed:
+                    OnWorkerDestroyed(pageWorkerDestroyed);
+                    break;
+                case PageDispatchMessageFromWorkerFirefoxEvent pageDispatchMessageFromWorker:
+                    OnDispatchMessageFromWorker(pageDispatchMessageFromWorker);
+                    break;
+            }
+        }
+
+        private void OnFrameAttached(PageFrameAttachedFirefoxEvent pageFrameAttached)
+            => Page.FrameManager.FrameAttached(pageFrameAttached.FrameId, pageFrameAttached.ParentFrameId);
+
+        private void OnNavigationCommitted(PageNavigationCommittedFirefoxEvent pageNavigationCommitted)
+        {
+            foreach (var pair in _workers)
+            {
+                if (pair.Key == pageNavigationCommitted.FrameId)
+                {
+                    OnWorkerDestroyed(new PageWorkerDestroyedFirefoxEvent { WorkerId = pair.Key });
+                }
+            }
+
+            Page.FrameManager.FrameCommittedNewDocumentNavigation(pageNavigationCommitted.FrameId, pageNavigationCommitted.Url, pageNavigationCommitted.Name ?? string.Empty, pageNavigationCommitted.NavigationId ?? string.Empty, false);
+        }
+
+        private void OnWorkerCreated(PageWorkerCreatedFirefoxEvent pageWorkerCreated)
+        {
+            string workerId = pageWorkerCreated.WorkerId;
+            var worker = new Worker(pageWorkerCreated.Url);
+            FirefoxSession tempWorkerSession = null;
+            var workerSession = new FirefoxSession(_session.Connection, "worker", workerId, async (id, request) =>
+            {
+                try
+                {
+                    await _session.SendAsync(new PageSendMessageToWorkerRequest
+                    {
+                        FrameId = pageWorkerCreated.FrameId,
+                        WorkerId = workerId,
+                        Message = new ConnectionRequest
+                        {
+                            Id = id,
+                            Method = request.Command,
+                            Params = request,
+                        }.ToJson(),
+                    }).ConfigureAwait(false);
+                }
+                catch (Exception e)
+                {
+                    tempWorkerSession.OnMessage(new ConnectionResponse

```

Comment:
```suggestion
                    tempWorkerSession?.OnMessage(new ConnectionResponse

```

just in case

Reply:
this will never really be null. I did this as a workaround since C# doesn't allow me to use the `workerSession` variable here since it's still being initialized. this will never throw due to a null reference exception